File: course/src/main/java/in/testpress/course/network/LeaderboardPager.java
package in.testpress.course.network;

import in.testpress.course.models.Reputation;
import in.testpress.models.TestpressApiResponse;
import in.testpress.network.BaseResourcePager;
import in.testpress.network.RetrofitCall;

public class LeaderboardPager extends BaseResourcePager<Reputation> {

    private TestpressCourseApiClient apiClient;

    public LeaderboardPager(TestpressCourseApiClient apiClient) {
        this.apiClient = apiClient;
    }

    public LeaderboardPager() {}

    public void setApiClient(TestpressCourseApiClient apiClient) {
        this.apiClient = apiClient;
    }

    @Override
    protected Object getId(Reputation resource) {
        return resource.getId();
    }

    @Override
    public RetrofitCall<TestpressApiResponse<Reputation>> getItems(int page, int size) {
        queryParams.put(TestpressCourseApiClient.PAGE, page);
        queryParams.put(TestpressCourseApiClient.PAGE_SIZE, 20);
        return apiClient.getLeaderboard(queryParams);
    }

}

File: course/src/main/java/in/testpress/course/pagers/LeaderboardPager.kt
package `in`.testpress.course.pagers

import `in`.testpress.course.api.TestpressCourseApiClient
import `in`.testpress.course.models.Reputation
import `in`.testpress.models.TestpressApiResponse
import `in`.testpress.network.BaseResourcePager
import retrofit2.Response

class LeaderboardPager(val apiClient: TestpressCourseApiClient) :
    BaseResourcePager<Reputation>() {

    override fun getId(resource: Reputation): Any =resource.id

    override fun getItems(page: Int, size: Int): Response<TestpressApiResponse<Reputation>> {
        queryParams.put(TestpressCourseApiClient.PAGE,page)
        queryParams.put(TestpressCourseApiClient.PAGE_SIZE, 20)
        return apiClient.getLeaderboard(queryParams).execute()
    }
}